clients/ts/signalr/tests/LongPollingTransport.test.ts

```diff
@@ -29,96 +29,95 @@ describe("LongPollingTransport", () => {
                 }
             })
             .on("DELETE", (r) => new HttpResponse(202));
-        const transport = new LongPollingTransport(client, null, NullLogger.instance, false, 100);
+        const transport = new LongPollingTransport(client, null, NullLogger.instance, false);
 
         await transport.connect("http://example.com", TransferFormat.Text);
-        await transport.stop();
+        const stopPromise = transport.stop();
 
         // This should complete within the shutdown timeout
         await pollCompleted.promise;
+
+        await stopPromise;
     });
 
-    it("sends DELETE request on stop", async () => {
+    it("204 server response stops polling and raises onClose", async () => {
         let firstPoll = true;
-        const deleteReceived = new PromiseSource();
-        const pollCompleted = new PromiseSource();
+        let onCloseCalled = false;
         const client = new TestHttpClient()
             .on("GET", async (r) => {
                 if (firstPoll) {
                     firstPoll = false;
                     return new HttpResponse(200);
                 } else {
-                    await deleteReceived.promise;
-                    // Force the shutdown timer to be registered by not returning inline
-                    await delay(10);
-                    pollCompleted.resolve();
+                    //
                     return new HttpResponse(204);
                 }
             })
-            .on("DELETE", (r) => {
-                deleteReceived.resolve();
-                return new HttpResponse(202);
-            });
-        const transport = new LongPollingTransport(client, null, NullLogger.instance, false, 100);
+            .on("DELETE", (r) => new HttpResponse(202));
```

Comment:
Should there be a `DELETE` request in this case? Maybe it's better to not expect it, since I don't think a `204` should result in a `DELETE` request (not 100% sure that the C# client behaves this way though, and if it sends the `DELETE` in the case of a `204`, the JS client should too)

Reply:
Yeah, we only call `SendDeleteRequest` if the sender is trigger for stopping in C#: https://github.com/aspnet/SignalR/blob/dev/src/Microsoft.AspNetCore.Http.Connections.Client/Internal/LongPollingTransport.cs#L106

Reply:
is copy pasta